### DISCUSSION TITLE
Logout with active_sessions feature does not invalidate JWT

### DISCUSSION THREAD

**Discussion by Samuelodan** [janko/rodauth-rails#238] [Q&A] [answered]
Hi there,
I'm trying to implement logout in an app, but the test indicate that the JWT still works after logout.

here's my test
```rb
    context 'Without Global logout' do
      it "invalidates the JWT" do
        account = create(:user)
        # login user
        post "/v1/auth/login", as: :json, params: {
          email: account.email,
          password: account.password
        }
        expect(response).to have_http_status(200)

        bearer_token = response.headers["Authorization"]

        # logout
        post "/v1/auth/logout", as: :json

        new_password = "newpassword"

        # attempt to change user password with same JWT (bearer_token)
        post "/v1/auth/change-password", as: :json, params: {
          password: account.password,
          new_password: new_password,
          password_confirmation: new_password
        }, headers: { Authorization: bearer_token }
        expect(response).to have_http_status(401)
        error_message = response.parsed_body.fetch("error")
        expect(error_message).to eq "Please login to continue"
      end
    end
```

The change password response comes back successful instead of failing.

I've added the feature to the config file. Is there anything I'm doing wrong?

**Comment by janko** [comment, marked as answer]
Are you calling `rodauth.check_active_session` in the `route` block? That's what actually checks whether the current session is active and doesn't let the request go through otherwise.

```rb
route do |r|
  rodauth(:v1).check_active_session
  r.rodauth(:v1)
end
```

I just added a [test](https://github.com/janko/rodauth-demo-rails/commit/d19eda23cd40f440b7b1163e3f2e79ae719d4fa4) to the official demo app, and it works just fine, logout does invalidate the previous token.

**Reply by Samuelodan** [reply]
Oh, interesting, I didn't know to call `check_active_session` in there. Thanks. I was starting to think I should try including an Authorization in the logout request.
How does the rodauth know the current session tho (especially with JWT)? I'm curious.

Edit: I just looked at your test and saw you added the header to the request. That makes sense to me now.

**Reply by Samuelodan** [reply]
One more question. If I wanted to logout globally, what value would I pass to the `global_logout` param? I couldn't find that info in Rodauth's Docs.

**Reply by janko** [reply]
You can provide any value for `global_logout`, Rodauth just checks that the param is present.

**Reply by Samuelodan** [reply]
I see. Thank you.
